app/assets/javascripts/admin/addon/components/schema-theme-setting/types/string.gjs

```diff
@@ -1,15 +1,76 @@
 import Component from "@glimmer/component";
 import { Input } from "@ember/component";
+import { tracked } from "@glimmer/tracking";
 import { on } from "@ember/modifier";
 import { action } from "@ember/object";
+import { and, not } from "truth-helpers";
+import concatClass from "discourse/helpers/concat-class";
+import I18n from "discourse-i18n";
+import FieldInputDescription from "admin/components/schema-theme-setting/field-input-description";
 
 export default class SchemaThemeSettingTypeString extends Component {
+  minLength = this.args.spec.validations?.min_length;
+  maxLength = this.args.spec.validations?.max_length;
+  required = this.args.spec.required;
+  @tracked touched = false;
+  @tracked value = this.args.value || "";
+
   @action
   onInput(event) {
-    this.args.onChange(event.currentTarget.value);
+    this.touched = true;
+    const newValue = event.currentTarget.value;
+    this.args.onChange(newValue);
+    this.value = newValue;
+  }
+
+  get validationErrorMessage() {
+    if (!this.touched) {
+      return;
+    }
+
+    const valueLength = this.value.length;
+
+    if (
+      (this.required && valueLength === 0) ||
+      (this.minLength && valueLength < this.minLength)
+    ) {
+      return I18n.t("admin.customize.theme.schema.fields.string.too_short", {
```

Comment:
Note that we don't display any message for because we have the `maxlength` attribute set on the input. That prevents the input value from exceeding the maxlength. Instead, a counter is shown when the `max_length` validation rule has been set.